pkg/security/secl/rules/policy.go

```diff
@@ -63,50 +63,108 @@ func applyOverride(rd1, rd2 *PolicyRule) {
 	// keep track of the combine
 	rd1.Def.Combine = rd2.Def.Combine
 
+	wasOverridden := false
 	// for backward compatibility, by default only the expression is copied if no options
 	if len(rd2.Def.OverrideOptions.Fields) == 0 {
 		rd1.Def.Expression = rd2.Def.Expression
-	} else if slices.Contains(rd2.Def.OverrideOptions.Fields, OverrideAllFields) {
+		wasOverridden = true
+	} else if slices.Contains(rd2.Def.OverrideOptions.Fields, OverrideAllFields) && rd1.Policy.Type == DefaultPolicyType {
+		tmpExpression := rd1.Def.Expression
 		*rd1.Def = *rd2.Def
+		rd1.Def.Expression = tmpExpression
+		wasOverridden = true
 	} else {
-		if slices.Contains(rd2.Def.OverrideOptions.Fields, OverrideExpressionField) {
-			rd1.Def.Expression = rd2.Def.Expression
-		}
 		if slices.Contains(rd2.Def.OverrideOptions.Fields, OverrideActionFields) {
-			rd1.Def.Actions = rd2.Def.Actions
+			for _, action := range rd2.Def.Actions {
+				duplicated := false
+				for _, a := range rd1.Actions {
+					if action.Equals(a.Def) {
+						duplicated = true
+						break
+					}
+				}
+				if !duplicated {
+					rd1.Def.Actions = append(rd1.Def.Actions, action)
+					wasOverridden = true
+				}
```

Comment:
are we sure we want to merge actions here (and not just override them) ? Same for tags